src/pages/index.js

```diff
@@ -20,88 +87,176 @@ const cardSection = new Section(
+function emptyCardLike(){
+  const likeBtn = document.querySelector('.card__like-button');
+  likeBtn.classList.remove('card__like-button--active');
+}
+
+// Fetch API cards and render them inside the same section
+api.getInitialCards()
+.then((cards) => {
+  cards.forEach((cardData) => {
+    const cardElement = createCard(cardData); // Generate card using createCard
+    cardSection.addItem(cardElement); // Properly add to `.gallery__cards`
+    
+    console.log(cardData.isLiked)
+    if(cardData.isLiked){
+      console.log(cardData._id);  
+      fillCardLike();
+    }else{
+      emptyCardLike();
+    }
+  });
+});
+
+//---------------------FORMS AND VALIDATION AND SUBMIT BUTTONS RESPONSES--------------------------
 
 // Initialize form validation
 const addCardValidator = new FormValidator(config, document.querySelector("#add__modal_form"));
 const profileValidator = new FormValidator(config, document.querySelector("#edit__modal_form"));
+const avatarValidator = new FormValidator(config, document.querySelector("#avatar__modal_form"));
 addCardValidator.enableValidation();
 profileValidator.enableValidation();
+avatarValidator.enableValidation();
 
 //-----------------------POPUPS--------------------------------
-
 // Create Popups
 const imagePopup = new PopupWithImage("#modal__picture");
 imagePopup.setEventListeners();
 
 //  Initialize `UserInfo`
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
-  jobSelector: ".profile__description"
+  jobSelector: ".profile__description",
+  avatarSelector: ".profile__avatar"
 });
 
 const profilePopup = new PopupWithForm("#profileModal", (formData) => {
-  userInfo.setUserInfo(formData);
+  userInfo.setUserInfo(formData); // Make changes on the page right away
+  
+  api.updateUserInfo(formData.title, formData.about) // sends to Api.js to save to database
+    .then((updatedData) => {
+      profilePopup.close(); 
+    })
+    .catch((err) => {
+      console.error("Error updating profile:", err);
+    });
 });
+
 profilePopup.setEventListeners();
 
+
 const addCardPopup = new PopupWithForm("#modal__add", (formData) => {
   const newCardData = {
     name: formData.name.trim(),
     link: formData.url.trim(),
   };
 
   if (newCardData.name && newCardData.link) {
-    cardSection.addItem(createCard(newCardData));
+    api.updateNewCards(newCardData) 
+      .then((savedCard) => {
+        // Ensure the savedCard has all necessary properties
+        const fullCardData = {
+          ...savedCard,
+          likes: [], // Initialize empty likes array
+          _id: savedCard._id, // Ensure _id is present
+        };
 
-    // Only reset form on submit
-    document.querySelector("#add__modal_form").reset();
-
-    // Close popup after adding a card
-    addCardPopup.close();
+        const cardElement = createCard(fullCardData)
+        cardSection.addItem(cardElement); 
+        document.querySelector("#add__modal_form").reset();
```

Comment:
`        document.querySelector("#add__modal_form").reset();
`

It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine